### DISCUSSION TITLE
Accessing responses in survey-html-form for conditional timeline

### DISCUSSION THREAD

**Discussion by dm0rg8n** [jspsych/jsPsych#781] [Q&A] [answered]
Hi!

I was just wondering if anyone could advise me on how to access the responses in a survey-html-form? I understand that I can use the conditional_function option to achieve this but I do not seem to be able to access the data. Essentially I have 6 statements that participants need to respond to, so that they can consent to taking part. I would like to write a conditional timeline which checks whether each of the responses given to those six statements are answered with a "yes" and if so - they can continue onto the next part of the experiment. If that is not the case they should be redirected elsewhere. Participants give there responses using drop down menus created using the select and option html tags.

I have tried using the following which does not work presumably because I am misunderstanding how the function indexes into the responses:

```
 var check_consent = {
    timeline: [informationsheet],
    conditional_function: function(){
        // get the data from the previous trial,
        // and check which option was selected
        var data = jsPsych.data.get().last(1).values()[0];
        if(data.button_pressed == "No"){
            return false;
        } else {
            return true;
        }
    }
}
```

But when I print the variable "data" to the console it shows the following:

```
rt: 11070.83000000057, responses: "{}", trial_type: "survey-html-form", trial_index: 1, time_elapsed: 12194, …}
internal_node_id: "0.0-1.0"
responses: "{}"
rt: 11070.83000000057
time_elapsed: 12194
trial_index: 1
trial_type: "survey-html-form"
__proto__: Object
```

So it appears that no responses are recorded and this may be why?

Can anyone advise on this? I would be super grateful! :D Sorry if this questions has already been answered and I missed it!

**Comment by vijaymarupudi** [comment]
Can you provide the html for the form? It is likely that you did not add the `name` attribute to the input elements.

**Comment by dm0rg8n** [comment, marked as answer]
Hi @vijaymarupudi,

Thank you for your fast response, I really appreciate it!

I have copied a condensed version of the html below, as the actual html is rather long and is a nightmare to look at! I can of course paste the full html here if you would like? But I figured this would be better. :D

```javascript
 var informationsheet = {
      type: 'survey-html-form',
      preamble: '<h1>Information Sheet and to Act as a Research Subject</h1> <h2>Study Title: <i>Picture Perfect Memory</i></h2><h3>Who is conducting the study? </p></body>',
      html: '<p>I have read and understood the information above</p><p><select id="consent1" required="required"><option value=""></option><option value="Yes">Yes</option><option value="No">No</option></select>'}; 
```

In the actual version there are of course six drop down menus, not just one. 

And yup, as you say i did not add the name attribute, what would you recommend?

Thanks!

**Reply by vijaymarupudi** [reply]
The `name` attribute to a form is required if you want to store data from it. It will be the key of the key-value object pairs returned by the plugin. For example, add the `name="consentSelect"` attribute to the `<select>` tag and you will see the answer in the data object.

**Reply by dm0rg8n** [reply]
Hi @vijaymarupudi,

That worked! I now have key-value pairs for all 7 questions which they have to answer. I now need to check that they have not responded "no" to any of the questions. I have done some searching for accessing values in the key value pairs but have had no success. Do you have any advice on how to access the values in the key "responses" and loop through to check the answers in the values: 

For example, printing the responses key of the data for the consent trial to the console produces this:

`responses: "{"consentSelect0":"Yes","consentSelect1":"Yes","consentSelect2":"Yes","consentSelect3":"Yes","consentSelect4":"Yes","consentSelect5":"Yes","consentSelect6":"Yes"}`"

And so I need to index into the values to check the responses. How would I go about doing this? It looks like the values themselves are already in a dictionary format so would it be easier to convert those into a dictionary and then access the values that way?

Thanks!

**Reply by vijaymarupudi** [reply]
To get that as a Javascript object (dictionary), use `JSON.parse(data.responses)`.
